auslib/admin/views/history.py

```diff
@@ -1,96 +1,108 @@
-            try:
-                value = json.dumps(value, indent=2, sort_keys=True)
-            except ValueError:
-                pass
-        elif value is None:
-            value = 'NULL'
-        elif isinstance(value, int):
-            value = unicode(str(value), 'utf8')
-        else:
-            value = unicode(value, 'utf8')
-        return value
-
-    def get(self, type_, change_id, field):
-        try:
-            value = self.get_value(type_, change_id, field)
-        except KeyError as msg:
-            self.log.warning("Bad input: %s", field)
-            return Response(status=400, response=str(msg))
-        except ValueError as msg:
-            return Response(status=404, response=str(msg))
-        value = self.format_value(value)
-        return Response(value, content_type='text/plain')
-
-
-class DiffView(FieldView):
-    """/diff/:type/:id/:field"""
-
-    def get_prev_id(self, value, change_id):
-
-        release_name = value['name']
-
-        table = dbo.releases.history
-        old_revision = table.select(
-            where=[
-                table.name == release_name,
-                table.change_id < change_id,
-                table.data_version != null()
-            ],
-            limit=1,
-            order_by=[table.timestamp.desc()],
-        )
-
-        return old_revision[0]['change_id']
-
-    def get(self, type_, change_id, field):
-        value = self.get_value(type_, change_id, field)
-        data_version = self.get_value(type_, change_id, "data_version")
-
-        prev_id = self.get_prev_id(value, change_id)
-        previous = self.get_value(type_, prev_id, field)
-        prev_data_version = self.get_value(type_, prev_id, "data_version")
-
-        value = self.format_value(value)
-        previous = self.format_value(previous)
-
-        result = difflib.unified_diff(
-            previous.splitlines(),
-            value.splitlines(),
-            fromfile="Data Version {}".format(prev_data_version),
-            tofile="Data Version {}".format(data_version),
-            lineterm=""
-        )
-
-        return Response('\n'.join(result), content_type='text/plain')
+from flask import Response, jsonify, request
+from sqlalchemy import and_
+from auslib.admin.views.base import HistoryAdminView
+
+
+class HistoryViewGetParameters(object):
+    def __init__(self,
+                 get_object_callback,
+                 history_filters_callback,
+                 process_revisions_callback,
+                 revisions_order_by,
+                 obj_not_found_msg='Requested object does not exist',
+                 response_key='revisions'):
+        self.get_object = get_object_callback
+        self.filters = history_filters_callback
+        self.process_revisions = process_revisions_callback
+        self.order_by = revisions_order_by
+        self.not_found_msg = obj_not_found_msg
+        self.response_key = response_key
+
+
+class HistoryViewPostParameters(object):
```

Comment:
I think it would be better to get rid of these and put the necessary parameters into get_revisions and revert_revision. Putting them here adds a layer of indirection that makes it more difficult to read the code. Some docstrings similar to what we have in AUSTable (to describe the parameters) would be greatly appreciated, too.